app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx

```diff
@@ -147,8 +235,15 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
   }
 
   handleLogin = () => {
-    this.props.login(this.props.publicKey)
+    if (this.state.publicKeys.length && this.state.address) {
+      const keyData = this.state.publicKeys.find(
```

Comment:
not sure if you considered this, but would it help if `publicKeys` was a hashmap instead?

Reply:
I think using an array is pretty simple and works fine but maybe I am missing something... using a hashmap was not something I considered

Reply:
I don't think we need a hashmap. there should not be a lot of public keys